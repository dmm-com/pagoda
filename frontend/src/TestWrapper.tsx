import { createTheme, ThemeProvider } from "@mui/material/styles";
import { SnackbarProvider } from "notistack";
import React, { FC } from "react";
import { MemoryRouter } from "react-router";

export const TestWrapper: FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  const theme = createTheme();
  return (
    <ThemeProvider theme={theme}>
      <SnackbarProvider maxSnack={1} autoHideDuration={100}>
        <MemoryRouter>{children}</MemoryRouter>
      </SnackbarProvider>
    </ThemeProvider>
  );
};

export const TestWrapperWithoutRoutes: FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  const theme = createTheme();
  return (
    <ThemeProvider theme={theme}>
      <SnackbarProvider maxSnack={1} autoHideDuration={100}>
        {children}
      </SnackbarProvider>
    </ThemeProvider>
  );
};
