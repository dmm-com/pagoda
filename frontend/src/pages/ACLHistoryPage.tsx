import { ACLObjtypeEnum } from "@dmm-com/airone-apiclient-typescript-fetch";
import AppsIcon from "@mui/icons-material/Apps";
import { Box, Container, IconButton } from "@mui/material";
import { styled } from "@mui/material/styles";
import React, { FC, useEffect, useState } from "react";

import { useAsyncWithThrow } from "../hooks/useAsyncWithThrow";

import { ACLHistoryList } from "components/acl/ACLHistoryList";
import { Loading } from "components/common/Loading";
import { PageHeader } from "components/common/PageHeader";
import { EntityBreadcrumbs } from "components/entity/EntityBreadcrumbs";
import { EntityControlMenu } from "components/entity/EntityControlMenu";
import { EntryBreadcrumbs } from "components/entry/EntryBreadcrumbs";
import { EntryControlMenu } from "components/entry/EntryControlMenu";
import { EntryImportModal } from "components/entry/EntryImportModal";
import { useTypedParams } from "hooks/useTypedParams";
import { aironeApiClient } from "repository/AironeApiClient";

const MenuBox = styled(Box)(({}) => ({
  width: "50px",
}));

export const ACLHistoryPage: FC = () => {
  const { objectId } = useTypedParams<{ objectId: number }>();
  const [breadcrumbs, setBreadcrumbs] = useState<JSX.Element>(<Box />);
  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
  const [openImportModal, setOpenImportModal] = React.useState(false);

  const acl = useAsyncWithThrow(async () => {
    return await aironeApiClient.getAcl(objectId);
  }, [objectId]);

  const aclHistory = useAsyncWithThrow(async () => {
    return await aironeApiClient.getAclHistory(objectId);
  }, [objectId]);

  const controlMenu = () => {
    if (acl.value == null) return;
    switch (acl.value.objtype) {
      case ACLObjtypeEnum.Entity:
        return (
          <EntityControlMenu
            entityId={acl.value.id}
            anchorElem={anchorEl}
            handleClose={() => setAnchorEl(null)}
            setOpenImportModal={setOpenImportModal}
          />
        );
      case ACLObjtypeEnum.Entry:
        if (acl.value.parent?.id) {
          return (
            <EntryControlMenu
              entityId={acl.value.parent.id}
              entryId={acl.value.id}
              anchorElem={anchorEl}
              handleClose={() => setAnchorEl(null)}
            />
          );
        }
    }
  };

  useEffect(() => {
    if (acl.value == null) return;

    switch (acl.value.objtype) {
      case ACLObjtypeEnum.Entity:
        aironeApiClient.getEntity(objectId).then((resp) => {
          setBreadcrumbs(
            <EntityBreadcrumbs entity={resp} title="ACL変更履歴" />,
          );
        });
        break;
      case ACLObjtypeEnum.Entry:
        aironeApiClient.getEntry(objectId).then((resp) => {
          setBreadcrumbs(<EntryBreadcrumbs entry={resp} title="ACL変更履歴" />);
        });
        break;
    }
  }, [acl.loading]);

  return (
    <Box className="container-fluid">
      {breadcrumbs}

      <PageHeader title={acl.value?.name ?? ""} description="ACL変更履歴">
        <MenuBox>
          <IconButton
            id="controlMenu"
            onClick={(e) => {
              setAnchorEl(e.currentTarget);
            }}
          >
            <AppsIcon />
          </IconButton>
          {controlMenu()}
        </MenuBox>
      </PageHeader>
      <EntryImportModal
        openImportModal={openImportModal}
        closeImportModal={() => setOpenImportModal(false)}
      />

      {aclHistory.loading ? (
        <Loading />
      ) : (
        <Container>
          <ACLHistoryList histories={aclHistory.value ?? []} />
        </Container>
      )}
    </Box>
  );
};
