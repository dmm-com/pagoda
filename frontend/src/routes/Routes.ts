export const basePath = "/ui/";

export const loginPath = () => "/auth/login/";
export const topPath = () => basePath;
export const advancedSearchPath = () => basePath + "advanced_search";
export const advancedSearchResultPath = () =>
  basePath + "advanced_search_result";
export const jobsPath = (targetId?: number) =>
  basePath + "jobs" + (targetId ? `?target_id=${targetId}` : "");
export const aclPath = (objectId: number | string) =>
  basePath + `acl/${objectId}`;
export const aclHistoryPath = (objectId: number | string) =>
  basePath + `acl/${objectId}/history`;
export const searchPath = () => basePath + "search";

// entries
export const newEntryPath = (entityId: number | string) =>
  basePath + `entities/${entityId}/entries/new`;
export const copyEntryPath = (
  entityId: number | string,
  entryId: number | string
) => basePath + `entities/${entityId}/entries/${entryId}/copy`;
export const entryDetailsPath = (
  entityId: number | string,
  entryId: number | string
) => basePath + `entities/${entityId}/entries/${entryId}/details`;
export const entryEditPath = (
  entityId: number | string,
  entryId: number | string
) => basePath + `entities/${entityId}/entries/${entryId}/edit`;
export const entityEntriesPath = (entityId: number | string) =>
  basePath + `entities/${entityId}/entries`;
export const restoreEntryPath = (entityId: number | string, keyword?: string) =>
  basePath +
  `entities/${entityId}/restore${keyword != null ? "?query=" + keyword : ""}`;
export const showEntryHistoryPath = (
  entityId: number | string,
  entryId: number | string
) => basePath + `entities/${entityId}/entries/${entryId}/history`;

// entities
export const entityHistoryPath = (entityId: number | string) =>
  basePath + `entities/${entityId}/history`;
export const listAliasPath = (entityId: number | string) =>
  basePath + `entities/${entityId}/alias`;
export const newEntityPath = () => basePath + "entities/new";
export const editEntityPath = (entityId: number | string) =>
  basePath + `entities/${entityId}`;
export const entitiesPath = () => basePath + "entities";

// triggers
export const newTriggerPath = () => basePath + "triggers/new";
export const editTriggerPath = (triggerId: number | string) =>
  basePath + `triggers/${triggerId}`;
export const triggersPath = () => basePath + "triggers";

// groups
export const newGroupPath = () => basePath + "groups/new";
export const groupPath = (groupId: number | string) =>
  basePath + `groups/${groupId}`;
export const groupsPath = () => basePath + "groups";

// users
export const newUserPath = () => basePath + "users/new";
export const userPath = (userId: number | string) =>
  basePath + `users/${userId}`;
export const usersPath = () => basePath + "users";

// roles
export const newRolePath = () => basePath + "roles/new";
export const rolePath = (roleId: number | string) =>
  basePath + `roles/${roleId}`;
export const rolesPath = () => basePath + "roles";


// category
export const newCategoryPath = () => basePath + "categories/new";
export const listCategoryPath = () => basePath + "categories/list";
export const editCategoryPath = (categoryId: number | string) => basePath + "categories/edit";