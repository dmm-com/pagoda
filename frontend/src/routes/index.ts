export * from "./AppRouter";
export * from "./Routes";
