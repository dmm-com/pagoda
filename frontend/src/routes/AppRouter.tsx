import { FC, ReactNode } from "react";
import {
  createBrowserRouter,
  createRoutesFromElements,
  Outlet,
  Route,
  RouterProvider,
  useRouteError,
} from "react-router";

import { ACLHistoryPage } from "../pages/ACLHistoryPage";
import { AliasEntryListPage } from "../pages/AliasEntryListPage";
import { CategoryListPage } from "../pages/CategoryListPage";
import { EntryCopyPage } from "../pages/EntryCopyPage";
import { EntryDetailsPage } from "../pages/EntryDetailsPage";
import { EntryRestorePage } from "../pages/EntryRestorePage";
import { NotFoundErrorPage } from "../pages/NotFoundErrorPage";
import { RoleEditPage } from "../pages/RoleEditPage";
import { RoleListPage } from "../pages/RoleListPage";

import { Header } from "components/common/Header";
import { ACLEditPage } from "pages/ACLEditPage";
import { AdvancedSearchPage } from "pages/AdvancedSearchPage";
import { AdvancedSearchResultsPage } from "pages/AdvancedSearchResultsPage";
import { CategoryEditPage } from "pages/CategoryEditPage";
import { DashboardPage } from "pages/DashboardPage";
import { EntityEditPage } from "pages/EntityEditPage";
import { EntityHistoryPage } from "pages/EntityHistoryPage";
import { EntityListPage } from "pages/EntityListPage";
import { EntryEditPage } from "pages/EntryEditPage";
import { EntryHistoryListPage } from "pages/EntryHistoryListPage";
import { EntryListPage } from "pages/EntryListPage";
import { GroupEditPage } from "pages/GroupEditPage";
import { GroupListPage } from "pages/GroupListPage";
import { JobListPage } from "pages/JobListPage";
import { LoginPage } from "pages/LoginPage";
import { TriggerEditPage } from "pages/TriggerEditPage";
import { TriggerListPage } from "pages/TriggerListPage";
import { UserEditPage } from "pages/UserEditPage";
import { UserListPage } from "pages/UserListPage";
import {
  aclHistoryPath,
  aclPath,
  advancedSearchPath,
  advancedSearchResultPath,
  copyEntryPath,
  editCategoryPath,
  editEntityPath,
  editTriggerPath,
  entitiesPath,
  entityEntriesPath,
  entityHistoryPath,
  entryDetailsPath,
  entryEditPath,
  groupPath,
  groupsPath,
  jobsPath,
  listAliasPath,
  listCategoryPath,
  loginPath,
  newCategoryPath,
  newEntityPath,
  newEntryPath,
  newGroupPath,
  newRolePath,
  newTriggerPath,
  newUserPath,
  restoreEntryPath,
  rolePath,
  rolesPath,
  showEntryHistoryPath,
  topPath,
  triggersPath,
  userPath,
  usersPath,
} from "routes/Routes";

// re-throw error to be caught by the root error boundary
const ErrorBridge: FC = () => {
  throw useRouteError();
};

interface Props {
  customRoutes?: {
    path: string;
    element: ReactNode;
  }[];
}

export const AppRouter: FC<Props> = ({ customRoutes }) => {
  const router = createBrowserRouter(
    createRoutesFromElements(
      <Route errorElement={<ErrorBridge />}>
        <Route path={loginPath()} element={<LoginPage />} />
        <Route
          path="/"
          element={
            <>
              <Header />
              <Outlet />
            </>
          }
        >
          {customRoutes &&
            customRoutes.map((r) => (
              <Route key={r.path} path={r.path} element={r.element} />
            ))}

          <Route path={newCategoryPath()} element={<CategoryEditPage />} />
          <Route
            path={editCategoryPath(":categoryId")}
            element={<CategoryEditPage />}
          />
          <Route path={listCategoryPath()} element={<CategoryListPage />} />
          <Route path={advancedSearchPath()} element={<AdvancedSearchPage />} />
          <Route
            path={advancedSearchResultPath()}
            element={<AdvancedSearchResultsPage />}
          />
          <Route path={newEntryPath(":entityId")} element={<EntryEditPage />} />
          <Route
            path={copyEntryPath(":entityId", ":entryId")}
            element={<EntryCopyPage />}
          />
          <Route
            path={entryDetailsPath(":entityId", ":entryId")}
            element={<EntryDetailsPage />}
          />
          <Route
            path={restoreEntryPath(":entityId")}
            element={<EntryRestorePage />}
          />
          <Route
            path={entryEditPath(":entityId", ":entryId")}
            element={<EntryEditPage />}
          />
          <Route
            path={showEntryHistoryPath(":entityId", ":entryId")}
            element={<EntryHistoryListPage />}
          />
          <Route
            path={entityEntriesPath(":entityId")}
            element={<EntryListPage />}
          />
          <Route
            path={entityHistoryPath(":entityId")}
            element={<EntityHistoryPage />}
          />
          <Route
            path={listAliasPath(":entityId")}
            element={<AliasEntryListPage />}
          />
          <Route path={newEntityPath()} element={<EntityEditPage />} />
          <Route
            path={editEntityPath(":entityId")}
            element={<EntityEditPage />}
          />
          <Route path={entitiesPath()} element={<EntityListPage />} />
          <Route path={newTriggerPath()} element={<TriggerEditPage />} />
          <Route
            path={editTriggerPath(":triggerId")}
            element={<TriggerEditPage />}
          />
          <Route path={triggersPath()} element={<TriggerListPage />} />
          <Route path={newGroupPath()} element={<GroupEditPage />} />
          <Route path={groupPath(":groupId")} element={<GroupEditPage />} />
          <Route path={groupsPath()} element={<GroupListPage />} />
          <Route path={jobsPath()} element={<JobListPage />} />
          <Route
            path={aclHistoryPath(":objectId")}
            element={<ACLHistoryPage />}
          />
          <Route path={aclPath(":objectId")} element={<ACLEditPage />} />
          <Route path={newUserPath()} element={<UserEditPage />} />
          <Route path={userPath(":userId")} element={<UserEditPage />} />
          <Route path={usersPath()} element={<UserListPage />} />
          <Route path={newRolePath()} element={<RoleEditPage />} />
          <Route path={rolePath(":roleId")} element={<RoleEditPage />} />
          <Route path={rolesPath()} element={<RoleListPage />} />
          <Route path={topPath()} element={<DashboardPage />} />
          <Route path="*" element={<NotFoundErrorPage />} />
        </Route>
      </Route>,
    ),
  );

  return <RouterProvider router={router} />;
};
