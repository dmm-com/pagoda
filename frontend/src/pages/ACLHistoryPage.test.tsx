/**
 * @jest-environment jsdom
 */

import { render, screen, act, waitFor } from "@testing-library/react";
import { http, HttpResponse } from "msw";
import { setupServer } from "msw/node";
import { createMemoryRouter, RouterProvider } from "react-router";

import { ACLHistoryPage } from "./ACLHistoryPage";

import { TestWrapperWithoutRoutes } from "TestWrapper";
import { aclHistoryPath } from "routes/Routes";

const server = setupServer(
  // getAcl
  http.get("http://localhost/acl/api/v2/acls/1", () => {
    return HttpResponse.json({
      name: "acl1",
      objtype: 1,
      is_public: false,
      default_permission: 1,
      parent: {
        id: 10,
        name: "Entity1",
      },
      roles: [
        {
          id: 1,
          name: "member1",
          description: "",
          current_permission: 1,
        },
      ],
    });
  }),
  // getAclHistory
  http.get("http://localhost/acl/api/v2/acls/1/history", () => {
    return HttpResponse.json([]);
  }),
  // getEntity
  http.get("http://localhost/entity/api/v2/1/", () => {
    return HttpResponse.json({
      id: 1,
      name: "test entity",
      note: "",
      isToplevel: false,
      hasOngoingChanges: false,
      attrs: [],
      webhooks: [],
    });
  }),
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

test("should match snapshot", async () => {
  Object.defineProperty(window, "django_context", {
    value: {
      user: {
        is_superuser: false,
      },
    },
    writable: false,
  });

  const router = createMemoryRouter(
    [
      {
        path: aclHistoryPath(":objectId"),
        element: <ACLHistoryPage />,
      },
    ],
    {
      initialEntries: [aclHistoryPath(1)],
    },
  );
  const result = await act(async () => {
    return render(<RouterProvider router={router} />, {
      wrapper: TestWrapperWithoutRoutes,
    });
  });
  await waitFor(() => {
    expect(screen.queryByTestId("loading")).not.toBeInTheDocument();
  });

  expect(result).toMatchSnapshot();
});
