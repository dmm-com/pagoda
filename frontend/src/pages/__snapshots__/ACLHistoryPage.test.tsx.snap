// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-1baykhj"
        >
          <div
            class="MuiBox-root css-t0pdf1"
          >
            <div
              class="MuiBox-root css-jqffe8"
            >
              <nav
                aria-label="breadcrumb"
                class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
              >
                <ol
                  class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
                >
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      data-discover="true"
                      href="/ui/"
                    >
                      Top
                    </a>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                  >
                    /
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      data-discover="true"
                      href="/ui/entities"
                    >
                      モデル一覧
                    </a>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                  >
                    /
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <div
                      class="MuiBox-root css-gvoll6"
                    >
                      <a
                        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                        data-discover="true"
                        href="/ui/entities/1/entries"
                      >
                        test entity
                      </a>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="LockIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                        />
                      </svg>
                    </div>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                  >
                    /
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
                    >
                      ACL変更履歴
                    </p>
                  </li>
                </ol>
              </nav>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1wo2fgg"
        >
          <div
            class="MuiBox-root css-t1gim7"
          >
            <div
              class="MuiBox-root css-9rtbay"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-178toxd-MuiTypography-root"
                id="title"
              >
                acl1
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
                id="description"
              >
                ACL変更履歴
              </h6>
              <div
                class="MuiBox-root css-1b3gh38"
              >
                <div
                  class="MuiBox-root css-vxcgou"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    id="controlMenu"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="AppsIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-yk10cl-MuiDivider-root"
            />
          </div>
        </div>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <table
              class="MuiTable-root css-rqglhn-MuiTable-root"
              id="table_history_list"
            >
              <thead
                class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head css-15pgmhf-MuiTableRow-root"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-flimy0-MuiTableCell-root"
                    scope="col"
                    width="200px"
                  />
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-flimy0-MuiTableCell-root"
                    scope="col"
                    width="540px"
                  >
                    <div
                      class="MuiBox-root css-k008qs"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-kwew1m-MuiTypography-root"
                      >
                        項目
                      </p>
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-kwew1m-MuiTypography-root"
                      >
                        変更前
                      </p>
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-kwew1m-MuiTypography-root"
                      >
                        変更後
                      </p>
                    </div>
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-flimy0-MuiTableCell-root"
                    scope="col"
                    width="140px"
                  >
                    実行日時
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-flimy0-MuiTableCell-root"
                    scope="col"
                    width="144px"
                  >
                    実行者
                  </th>
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
              />
            </table>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid MuiBox-root css-0"
    >
      <div
        class="MuiBox-root css-1baykhj"
      >
        <div
          class="MuiBox-root css-t0pdf1"
        >
          <div
            class="MuiBox-root css-jqffe8"
          >
            <nav
              aria-label="breadcrumb"
              class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
            >
              <ol
                class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
              >
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <a
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    data-discover="true"
                    href="/ui/"
                  >
                    Top
                  </a>
                </li>
                <li
                  aria-hidden="true"
                  class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                >
                  /
                </li>
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <a
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    data-discover="true"
                    href="/ui/entities"
                  >
                    モデル一覧
                  </a>
                </li>
                <li
                  aria-hidden="true"
                  class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                >
                  /
                </li>
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <div
                    class="MuiBox-root css-gvoll6"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      data-discover="true"
                      href="/ui/entities/1/entries"
                    >
                      test entity
                    </a>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="LockIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                      />
                    </svg>
                  </div>
                </li>
                <li
                  aria-hidden="true"
                  class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                >
                  /
                </li>
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
                  >
                    ACL変更履歴
                  </p>
                </li>
              </ol>
            </nav>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1wo2fgg"
      >
        <div
          class="MuiBox-root css-t1gim7"
        >
          <div
            class="MuiBox-root css-9rtbay"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-178toxd-MuiTypography-root"
              id="title"
            >
              acl1
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
              id="description"
            >
              ACL変更履歴
            </h6>
            <div
              class="MuiBox-root css-1b3gh38"
            >
              <div
                class="MuiBox-root css-vxcgou"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  id="controlMenu"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="AppsIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-yk10cl-MuiDivider-root"
          />
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <table
            class="MuiTable-root css-rqglhn-MuiTable-root"
            id="table_history_list"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-15pgmhf-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-flimy0-MuiTableCell-root"
                  scope="col"
                  width="200px"
                />
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-flimy0-MuiTableCell-root"
                  scope="col"
                  width="540px"
                >
                  <div
                    class="MuiBox-root css-k008qs"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-kwew1m-MuiTypography-root"
                    >
                      項目
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-kwew1m-MuiTypography-root"
                    >
                      変更前
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-kwew1m-MuiTypography-root"
                    >
                      変更後
                    </p>
                  </div>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-flimy0-MuiTableCell-root"
                  scope="col"
                  width="140px"
                >
                  実行日時
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-flimy0-MuiTableCell-root"
                  scope="col"
                  width="144px"
                >
                  実行者
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            />
          </table>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
